test/regression/sql/query_filter.sql

```diff
@@ -0,0 +1,12 @@
+CREATE TABLE query_filter_int(a INT);
+INSERT INTO query_filter_int SELECT g FROM generate_series(1,100) g;
+SELECT COUNT(*) FROM query_filter_int WHERE a  <= 50;
+DROP TABLE query_filter_int;
+
+CREATE TABLE query_filter_float(a FLOAT8);
+INSERT INTO query_filter_float VALUES (0.9), (1.0), (1.1);
+SELECT COUNT(*) = 1 FROM query_filter_float WHERE a < 1.0;
+SELECT COUNT(*) = 2 FROM query_filter_float WHERE a <= 1.0;
+SELECT COUNT(*) = 2 FROM query_filter_float WHERE a < 1.1;
```

Comment:
In my experience you should generally avoid returning booleans from queries in pgregress tests, because it makes it harder to debug the test when it starts failing. With your current code you only know that the test failed, but not why. With the following, you could see how many rows were found, which can be helpful with debugging. 
```suggestion
SELECT COUNT(*) FROM query_filter_float WHERE a < 1.0;
SELECT COUNT(*) FROM query_filter_float WHERE a <= 1.0;
SELECT COUNT(*) FROM query_filter_float WHERE a < 1.1;
```